src/deluge/dsp/filter/filter.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2015-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "definitions_cxx.hpp"
+#include "deluge/dsp/stereo_sample.h"
+#include "util/functions.h"
+#include <cstdint>
+namespace deluge::dsp::filter {
+// Filter Interface
+template <typename T>
+class Filter {
+public:
+	Filter() = default;
+	//returns a gain compensation value
+	q31_t configure(q31_t frequency, q31_t resonance, LPFMode lpfMode, q31_t filterGain) {
```

Comment:
Frequency is always treated as 0-1 in digital filtering, where 1 is the maximum possible cutoff. In this case it's 0-44100 but that's just because of the sample rate we choose to output at

Similarly resonance is neutral at 0, resonance is loosely the gain on a feedback path so that's also somewhat self explanatory. 0 is no resonance and 1 is max resonance